lib/uefi/thread_utils.h

```diff
@@ -0,0 +1,28 @@
+
+#ifndef __LIB_UEFI_THREAD_UTILS_H_
+#define __LIB_UEFI_THREAD_UTILS_H_
+
+#include <kernel/thread.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#ifdef __cplusplus
+
+template <typename Func>
+int functor_thread_trampoline(void *f) {
+  auto callable = reinterpret_cast<Func *>(f);
+  int ret = (*callable)();
+  delete callable;
+  return ret;
+}
+
+template <typename Func>
+thread_t *thread_create(const char *name, Func f, int priority,
```

Comment:
might be less confusing to name this the same as the underlying C version.

Reply:
The underlying C version is also called `thread_create` , this is what the C++ version is named. I don't see what needs to change.

Reply:
Oh I thought it was just a bit confusing to name a new C++ overriding function that same name as the underlying C one.

Reply:
Done, renamed to thread_create_functor